cirq/experiments/fidelity_estimation.py

```diff
@@ -642,3 +646,195 @@ def per_cycle_depth(df):
+        `xxx_default` attributes, s is `initial_simplex_step_size` and `basis_vec`
+        is a one-hot encoded vector for each parameter for which the `parameterize_xxx`
+        attribute is True.
+
+        We also return a list of parameter names so the Cirq `param_resovler`
+        can be accurately constructed during optimization.
+        """
+        x0 = []
+        names = []
+        if self.parameterize_theta:
+            x0 += [self.theta_default]
+            names += [THETA.name]
+        if self.parameterize_zeta:
+            x0 += [self.zeta_default]
+            names += [ZETA.name]
+        if self.parameterize_chi:
+            x0 += [self.chi_default]
+            names += [CHI.name]
+        if self.parameterize_gamma:
+            x0 += [self.gamma_default]
+            names += [GAMMA.name]
+        if self.parameterize_phi:
+            x0 += [self.phi_default]
+            names += [PHI.name]
+
+        x0 = np.asarray(x0)
+        n_param = len(x0)
+        initial_simplex = [x0]
+        for i in range(n_param):
+            basis_vec = np.eye(1, n_param, i)[0]
+            initial_simplex += [x0 + initial_simplex_step_size * basis_vec]
+        initial_simplex = np.asarray(initial_simplex)
+
+        return initial_simplex, names
+
+
+@dataclass(frozen=True)
+class SqrtISwapXEBOptions(XEBPhasedFSimCalibrationOptions):
+    """Options for calibrating a sqrt(ISWAP) gate using XEB.
+
+    As such, the default for theta is changed to -pi/4 and the parameterization
+    predicate seeks out sqrt(ISWAP) gates.
+    """
+
+    theta_default: float = -np.pi / 4
+
+    @staticmethod
+    def should_parameterize(op: 'cirq.Operation') -> bool:
+        return op.gate == SQRT_ISWAP
+
+
+def parameterize_phased_fsim_circuit(
+    circuit: 'cirq.Circuit',
+    phased_fsim_options: XEBPhasedFSimCalibrationOptions,
+) -> 'cirq.Circuit':
+    """Parameterize PhasedFSim-like gates in a given circuit according to
+    `phased_fsim_options`.
+    """
+    options = phased_fsim_options
+    theta = THETA if options.parameterize_theta else options.theta_default
+    zeta = ZETA if options.parameterize_zeta else options.zeta_default
+    chi = CHI if options.parameterize_chi else options.chi_default
+    gamma = GAMMA if options.parameterize_gamma else options.gamma_default
+    phi = PHI if options.parameterize_phi else options.phi_default
+
+    new_moments = []
+    for moment in circuit.moments:
+        new_ops = []
+        for op in moment.operations:
+            if options.should_parameterize(op):
+                new_ops += [
+                    ops.PhasedFSimGate(
+                        theta=theta,
+                        zeta=zeta,
+                        chi=chi,
+                        gamma=gamma,
+                        phi=phi,
+                    ).on(*op.qubits)
+                ]
+            else:
+                new_ops += [op]
+        new_moments += [ops.Moment(new_ops)]
```

Comment:
I would be tempted to rewrite this as a nested list-generator expression. Up to you.

Reply:
Some people don't like gnarly list comprehensions. I think it could look nicer if I were allowed to format it:

```python
    circuit = Circuit([
        ops.Moment([
            ops.PhasedFSimGate(theta=theta, zeta=zeta, chi=chi, gamma=gamma, phi=phi).on(*op.qubits)
            if options.should_parameterize(op)
            else op for op in moment.operations
        ]) for moment in circuit.moments
    ])
```

Under the black formatter:

```python
    circuit = Circuit(
        [
            ops.Moment(
                [
                    ops.PhasedFSimGate(theta=theta, zeta=zeta, chi=chi, gamma=gamma, phi=phi).on(
                        *op.qubits
                    )
                    if options.should_parameterize(op)
                    else op
                    for op in moment.operations
                ]
            )
            for moment in circuit.moments
        ]
    )

```


Without comprehensions:
```python
    new_moments = []
    for moment in circuit.moments:
        new_ops = []
        for op in moment.operations:
            if options.should_parameterize(op):
                new_ops += [
                    ops.PhasedFSimGate(
                        theta=theta,
                        zeta=zeta,
                        chi=chi,
                        gamma=gamma,
                        phi=phi,
                    ).on(*op.qubits)
                ]
            else:
                new_ops += [op]
        new_moments += [ops.Moment(new_ops)]
    circuit = Circuit(new_moments)
```

@balopat or others: do you have a preference for style here?

Reply:
This doesn't answer the style question, but both Circuit and Moment accept raw comprehensions (i.e. no brackets):
```python
circuit = Circuit(
    ops.Moment(
        ops.PhasedFSimGate(theta=theta, zeta=zeta, chi=chi, gamma=gamma, phi=phi).on(
            *op.qubits
        )
        if options.should_parameterize(op)
        else op
        for op in moment.operations
    )
    for moment in circuit.moments
)
```
Doing this makes the `black` formatting a little less egregious.

Reply:
+1 from me for raw comprehensions. In this particular case, you could also clean it up by reusing the gate:
```python
fsim_gate = ops.PhaseFSimGate(theta=theta, zeta=zeta, chi=chi, gamma=gamma, phi=phi)

circuit = Circuit(
    ops.Moment(
        fsim_gate.on(*op.qubits) if options.should_parameterize(op) else op
        for op in moment.operations
    )
    for moment in circuit.moments
)
```
It might also be possible to wrap up that conditional expression into a method on the `XEBPhasedFSimCharacterizationOptions` class:
```python
circuit = Circuit(
    ops.Moment(
        options.parameterize(op) for op in moment.operations
    )
    for moment in circuit.moments
)
```
Finally, as a general comment, this pattern is common enough that we might consider adding a `map_op` method on `cirq.Circuit` that takes a function from op -> op.

Reply:
Yupp, +1 for list comprehension, it's more concise and for me more readable. I also really like @maffoo's comment on moving the responsibility of parametrization to the option. 

Re: `map_op` - it feels like then we're entering "compiler" territory with the circuit. While it may seem like a convenient syntactical sugar, it would be a bit surprising to me, as it is not the circuit's responsibility to transform itself.

Reply:
tell that to `Circuit.transform_qubits`

Reply:
Thanks all for the advice!